lib/utils/aws/aws_test.go

```diff
@@ -255,3 +255,140 @@ func TestValidateRoleARNAndExtractRoleName(t *testing.T) {
 		})
 	}
 }
+
+func TestParseRoleARN(t *testing.T) {
+	t.Parallel()
+	tests := map[string]struct {
+		arn             string
+		wantErrContains string
+	}{
+		"valid role arn": {
+			arn: "arn:aws:iam::123456789012:role/test-role",
+		},
+		"arn fails to parse": {
+			arn:             "foobar",
+			wantErrContains: "invalid AWS ARN",
+		},
+		"sts arn is not iam": {
+			arn:             "arn:aws:sts::123456789012:federated-user/Alice",
+			wantErrContains: "not an AWS IAM role",
+		},
+		"iam arn is not a role": {
+			arn:             "arn:aws:iam::123456789012:user/test-user",
+			wantErrContains: "not an AWS IAM role",
+		},
+		"iam role arn is missing role name": {
+			arn:             "arn:aws:iam::123456789012:role",
+			wantErrContains: "missing AWS IAM role name",
+		},
+	}
+
+	for name, tt := range tests {
+		tt := tt
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
```

Comment:
nit: IMHO these quick tests don't benefit much from parallel tests.

Reply:
👍 good point, I parallelized them since they were trivial but it makes no difference for runtime. will remove